### ISSUE TITLE
`no-member` false negative if member conditionally defined on constant false value

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#5551]
I also wonder what happens when the assignment in the base class is conditional.
Something like:
```python
class ClassWithMember:
     """Member defined in superclass."""
     def __init__(self):
         if False:
             self.member = True

class AssignMemberFromSuper(ClassWithMember):
     """This assignment is valid due to inheritance."""
     def __init__(self):
         super().__init__()
         self.member = self.member
```
That would require control flow, but would we emit the message with the current code? Or do we assume the assignment succeeded?
If we still emit we might want to open an issue about it to track future progress.

_Originally posted by @DanielNoord in https://github.com/PyCQA/pylint/pull/5544#r771456956_
